include/Graphs/VFG.h

```diff
@@ -357,19 +357,19 @@ class VFG : public GenericVFGTy
     /// Connect VFG nodes between caller and callee for indirect call site
     //@{
     /// Connect actual-param and formal param
-    virtual inline void connectAParamAndFParam(const PAGNode* cs_arg, const PAGNode* fun_arg, const CallBlockNode*, CallSiteID csId, VFGEdgeSetTy& edges)
+    virtual inline void connectAParamAndFParam(const PAGNode* csArg, const PAGNode* funArg, const CallBlockNode* cbn, CallSiteID csId, VFGEdgeSetTy& edges)
     {
-        NodeID actualParam = getDef(cs_arg);
-        NodeID formalParam = getDef(fun_arg);
+        NodeID actualParam = getActualParmVFGNode(csArg, cbn)->getId();
```

Comment:
How about if the AcutalParmVFGNode was opt out (removed) during SVFGOPT?

Reply:
I think this change will only work for indirect callsites but not normally direct callsites. We will need to add a conditional check.

Reply:
> How about if the AcutalParmVFGNode was opt out (removed) during SVFGOPT?

For the first comment, this shouldn't be run by `SVFGOPT` because it overrides this function, so `SVFGOPT` will keep doing whatever it was doing before.

> I think this change will only work for indirect callsites but not normally direct callsites. We will need to add a conditional check.

Isn't this only called for indirect calls anyway on the fly?

The VFG construction looks okay to me, in that `getActualParmVFGNode` is used to make actual param -> formal param edges.
```
        else if(FormalParmVFGNode* formalParm = SVFUtil::dyn_cast<FormalParmVFGNode>(node))
        {
            for(CallPESet::const_iterator it = formalParm->callPEBegin(), eit = formalParm->callPEEnd();
                    it!=eit; ++it)
            {
                const CallBlockNode* cs = (*it)->getCallSite();
                ActualParmVFGNode* acutalParm = getActualParmVFGNode((*it)->getSrcNode(),cs);
                addInterEdgeFromAPToFP(acutalParm,formalParm,getCallSiteID(cs, formalParm->getFun()));
            }
        }
```

Reply:
True. Only used when on the fly build. Could you add an assertion in `connectAParamAndFParam` to make sure the Formal and actual SVFGNodes do exist and have not been removed from SVFG? 

Though it fixed the problem, It still not quite clear why `getDef` won't work. (1) The formal parameter once removed will be replaced with a Phi node and its definition site will be redefined to point to that Phi (https://github.com/SVF-tools/SVF/blob/master/include/Graphs/SVFGOPT.h#L238) and (2) the actual parameter once removed will have the same value as its definition site too.

Reply:
I'm not sure if such an assertion would be meaningful because SVFG node removal only removes the edges, not the actual node itself (assuming it even happens in `SVFG`).

>Though it fixed the problem ...

The changes should only apply to `SVFG` since `SVFGOPT` overrides this function with one which considers phi.

Reply:
Actually, `getActualParmVFGNode` and the other 3 similar functions have an assert in them, so I think it would be unnecessary to assert.